controllers/clusterca_to_secret.go

```diff
@@ -0,0 +1,190 @@
+package controllers
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	Namespace  = "global-trust-certificates"
+	SecretName = "cluster-ca-secret"
+)
+
+func ClusterCaToSecret(client client.Client) {
+	log := log.Log.WithName("ClusterCAToSecret")
+
+	ticker := time.NewTicker(20 * time.Second)
+	quit := make(chan struct{})
+	for {
+		select {
+		case <-ticker.C:
+
+			isPresent, err := checkIfSecretPresent(context.Background(), client, log)
+
+			if isPresent {
+				// check the hash of the secret data with cert.
+				sec, err := getSecret(context.Background(), client, log)
+				if err != nil {
+					log.Error(err, "Failed to get secret")
+				}
+
+				clusterCert, err := getCert(log)
+				if err != nil {
+					log.Error(err, "Failed to get cluster CA")
+				}
+
+				clusterCAPemCertHash, err := getPemCertificateHash(*clusterCert, log)
+				if err != nil {
+					log.Error(err, "Failed to get cluster CA Hash")
+				}
+
+				if clusterCAPemCertHash == getCertificateHashFromSecret(*sec) {
```

Comment:
You may want to use bytes.Equal(a, b) when comparing byte slices as using double-equals will lead to inconsistent behavior due to difference in how slices are compared vs arrays.

Reply:
So the certificatePackage controller also compares the hashes this way:
`return expectedHash == getCertificateHashFromSecret(*secret), nil`

Also there is a certhasher function that was already present which I used to Hash both secret and CA and encode them into a String. Then comparing those. Should I not encode to string?
`return hex.EncodeToString(hash), nil`